onnx/defs/schema.h

```diff
@@ -33,6 +33,15 @@ typedef std::unordered_set<DataType> DataTypeSet;
 typedef std::unordered_map<std::string, std::pair<DataTypeSet, std::string>>
     TypeConstraintMap;
 
+typedef TensorShapeProto_Dimension InferenceDimension;
+
+struct InferenceContext {
+  virtual const AttributeProto* getAttribute(std::string name) = 0;
+  virtual const std::vector<TypeProto_Tensor>& getAllInputTypes() = 0;
```

Comment:
Great! Personal suggestion: also put the output typeproto as an interface here?

Reply:
ok, will do